synapse/daemon.py

```diff
@@ -466,7 +466,8 @@ def _getTaskFiniMesg(self, task, valu):
                         logger.exception('Error running %r', valu)
                     finally:
                         await valu.fini()
-                s_glob.plex.coroLoopTask(spinshareloop())
+                task = s_glob.plex.coroLoopTask(spinshareloop())
+                self.onfini(task.cancel)
```

Comment:
this call needs to happen from the ioloop since cancel is not threadsafe

Reply:
`self.onfini(functools.partial(s_glob.plex.callSoonSafe, task.cancel))`?